pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -366,40 +368,30 @@ func (r *ReconcileHyperConverged) getHyperConverged(req *common.HcoRequest) (*hc
 }
 
 // updateHyperConverged updates the HyperConverged resource according to its state in the request.
-func (r *ReconcileHyperConverged) updateHyperConverged(request *common.HcoRequest) error {
+func (r *ReconcileHyperConverged) updateHyperConverged(request *common.HcoRequest) (bool, error) {
 
 	// Since the status subresource is enabled for the HyperConverged kind,
 	// we need to update the status and the metadata separately.
 	// Moreover, we need to update the status first, in order to prevent a conflict.
 	// In addition, spec changes are removed by status update, but since status update done first, we need
 	// to store the spec and recover it after status update
 
-	var spec hcov1beta1.HyperConvergedSpec
 	if request.Dirty {
-		request.Instance.Spec.DeepCopyInto(&spec)
+		err := r.updateHyperConvergedSpecMetadata(request)
+		if err != nil {
+			r.logHyperConvergedUpdateError(request, err, "Failed to update HCO CR")
+			return false, err
```

Comment:
why not requeue in case the spec/metadata failed to be updated?

Reply:
look t line 228. We're queuing only in update conflict, in order to avoid endless loop, if something went terribly wrong.

Reply:
and if you have something to update in this iteration that was failed not due to conflict error, are we dismissing this change?

Reply:
Yes (and this is not a change. This is the current behavior)

Reply:
just wanted to be sure. thanks.
/approve
/lgtm